### ISSUE TITLE
False positive prop-types for multi-comp components

### ISSUE THREAD

**Issue by mbarlock** [yannickcr/eslint-plugin-react#267]
Although generally bad practice its needed sometimes. For example, when using [dynamic redux-forms](https://github.com/erikras/redux-form#dynamic-forms)

```javascript
import React, {Component, PropTypes} from "react";
import {connectReduxForm} from "redux-form";

class DynamicForm extends Component {
    render () {
        const {formName, fieldsNeeded} = this.props;

        // define form class
        class Form extends Component {
            render () {
                const {fields} = this.props;  // passed in by redux-form

                return (
                    <div>
                        {Object.keys(fields).map(fieldName =>
                            <div key={fieldName}>
                                <label>{fieldName}</label>
                                <input type="text" {...fields[fieldName]}/>
                            </div>
                        )}
                    </div>
                );
            }
        }

        Form.propTypes = {
            fields: PropTypes.object.isRequired
        };

        // connect Form to Redux and decorate with redux-form
        Form = connectReduxForm({ form: formName, fields: fieldsNeeded })(Form);

        // render connected and decorated form
        return <Form/>;
    }
}

DynamicForm.propTypes = {
    formName: PropTypes.string.isRequired,
    fieldsNeeded: PropTypes.arrayOf(PropTypes.string).isRequired
};
```

yields 

```
Message:
    'fields' is missing in props validation for DynamicForm.render.Form
Details:
    fileName: /Users/michael/workspace/paddle/src/scripts/components/ToolMetadataForm.js
    lineNumber: 11
```

**Comment by yannickcr** [comment]
Wow, first time I see this. Yep this is not supported at the time.

**Comment by SpainTrain** [comment]
:+1: Thanks for fixing this!  Will this drop in a new version soon-ish?  :heart:
